### PR TITLE
Mission - Alter and fix functions

### PR DESCRIPTION
**Pull request by Mike-MF**
**Requires (#675)**

**When merged this pull request will:**
- Converts all AL/EG commands to events and utilises targetEvent where necessary.
- respiratorEffects now has optional 3rd param (see attached images) and utilises 2.16 cutRsc `drawOverHud` param. (draw under)
- Bomber & Hunt now use new players function.
- Alters Hunt function; Increases refresh rate to 10 by default, removes `setGroupOwner`
- Fixes Bomber function; removed `setGroupOwner`, moved `setSpeedMode` to after `doMove` call, removed the delay on exploding after unconscious.

**Testing:**
- Tested bomber on the server
- Tested respiratorEffects on the server
- Hunt uses the same logic as Bomber, should work.

<details>
<summary>Images</summary>

Without additional effect.
Before

With.
After

</details>

### CODE REVIEW

File: addons/mission/functions/fnc_respiratorEffects.sqf [hunk 1]
```diff
@@ -81,4 +82,4 @@ if (isNil QGVAR(respiratorMasks)) then {
     };
 
     GVAR(oldGlasses) = _goggles;
-} , 1, [_markers, _damagePerTick, _damageTickRate]] call CBA_fnc_addPerFrameHandler;
+} , 1, [_damagePerTick, _damageTickRate]] call CBA_fnc_addPerFrameHandler;
```
Comment (BrettMayson) [inline addons/mission/functions/fnc_respiratorEffects.sqf:85]:
```suggestion
}, 1, [_damagePerTick, _damageTickRate]] call CBA_fnc_addPerFrameHandler;
```

**Review by BrettMayson — DISMISSED** [review]

**Review by jonpas — COMMENTED** [review]
This confuses me. Why are we removing auto-switch back to server? Why not target execute those functions appropriately instead of blacklisting or auto-switching at all?

File: addons/mission/functions/fnc_bomber.sqf [hunk 2]
```diff
@@ -24,14 +24,12 @@ params ["_unit", ["_detonateRadius", 10], ["_activateDistance", 100], ["_screami
 
 if (!is3DENPreview && {!isServer}) exitWith {};
 
-// Add vest, blacklist & set server as owner
+// Add vest
 _unit addVest "Umi_Bomb_Vest_Camo";
 
+// Should happen before any HC transfer.
```
Comment (jonpas) [inline addons/mission/functions/fnc_bomber.sqf:30]: Does it though? What if I call this function half an hour into the mission?

Comment (Mike-MF) [reply addons/mission/functions/fnc_bomber.sqf:30]: Then it will fail.

**Comment by Mike-MF** [comment]
> This confuses me. Why are we removing auto-switch back to server? Why not target execute those functions appropriately instead of blacklisting or auto-switching at all?

The auto-switch is what was breaking it, AI would not move and would stand in place, screaming loudly. I just prefer just doing the simple solution of HC blacklist and running it.

File: addons/mission/functions/fnc_bomber.sqf [hunk 3]
```diff
@@ -24,15 +24,13 @@ params ["_unit", ["_detonateRadius", 10], ["_activateDistance", 100], ["_screami
...
 _unit setVariable ["acex_headless_blacklist", true, true];
-(group _unit) setGroupOwner 2;
 
-// Set unit speed & disable relevant AI
-_unit setSpeedMode "FULL";
-_unit setSpeaker "ACE_NoVoice";
+["ace_common_setSpeaker", [_unit, "ACE_NoVoice"]] call CBA_fnc_globalEvent;
```
Comment (jonpas) [inline addons/mission/functions/fnc_bomber.sqf:33]:
```suggestion
[QACEGVAR(common,setSpeaker), [_unit, "ACE_NoVoice"]] call CBA_fnc_globalEvent;
```

File: addons/mission/functions/fnc_bomber.sqf [hunk 4]
```diff
@@ -46,31 +44,28 @@ private _randomExplosive = selectRandom ["DemoCharge_Remote_Ammo_Scripted", "Sat
...
+        private _players = (true call FUNC(players)) select {(_unit distance _x) < _activateDistance};
 
         if (_players isNotEqualTo []) then {
             _args set [5,  selectRandom _players];
         };
     };
 
     if (CBA_missionTime >= _time + 5) then {
-        _unit doMove (position _nearest);
+        [QGVAR(doMove), [_unit, (position _nearest)], _unit] call CBA_fnc_targetEvent;
```
Comment (jonpas) [inline addons/mission/functions/fnc_bomber.sqf:55]:
```suggestion
        [QGVAR(doMove), [_unit, position _nearest], _unit] call CBA_fnc_targetEvent;
```

File: addons/mission/functions/fnc_bomber.sqf [hunk 5]
```diff
@@ -46,31 +44,28 @@ private _randomExplosive = selectRandom ["DemoCharge_Remote_Ammo_Scripted", "Sat
...
+        _unit setSpeedMode "FULL";
```
Comment (jonpas) [inline addons/mission/functions/fnc_bomber.sqf:56]: Requires `targetEvent` as well.

File: addons/mission/functions/fnc_bomber.sqf [hunk 6]
```diff
@@ -46,31 +44,28 @@ private _randomExplosive = selectRandom ["DemoCharge_Remote_Ammo_Scripted", "Sat
     _args params ["_unit", "_detonateRadius", "_activateDistance", "_screamingDistance", "_time", "_nearest", "_randomExplosive"];
 
     if (isNull _nearest) exitWith {
-        private _players = (call CBA_fnc_players) select {
-            isTouchingGround _x && {(_unit distance _x) < _activateDistance}
-        };
+        private _players = (true call FUNC(players)) select {(_unit distance _x) < _activateDistance};
```
Comment (jonpas) [inline addons/mission/functions/fnc_bomber.sqf:47]:
```suggestion
        private _players = ([true] call FUNC(players)) select {(_unit distance _x) < _activateDistance};
```
You might add more arguments to `FUNC(players)` in the future.

**Review by jonpas — CHANGES_REQUESTED** [review]

**Comment by Mike-MF** [comment]
- targetEvent commit should cover all AL/EG commands where locality is an issue.
- removed blacklist checks from functions.

**Comment by Mike-MF** [comment]
I tested all of them on the server today, 2 tests of each, one active from mission start (with HC transfer delay) and one started mid-mission.

All work as expected.

**Review by jonpas — APPROVED** [review]

**Pull request merged**